pkg/kgo/compression.go

```diff
@@ -269,21 +268,25 @@ type zstdDecoder struct {
 	inner *zstd.Decoder
 }
 
-func (d *decompressor) decompress(src []byte, codec byte) ([]byte, error) {
+func (d *decompressor) decompress(src []byte, codec byte, pool *pool.BucketedPool[byte]) ([]byte, error) {
 	// Early return in case there is no compression
 	compCodec := codecType(codec)
 	if compCodec == codecNone {
 		return src, nil
 	}
-
-	out := decodedBuffers.Get().(*bytes.Buffer)
-	out.Reset()
-	defer func() {
-		if out.Cap() > maxPoolDecodedBufferSize {
-			return // avoid keeping large buffers in the pool
-		}
-		decodedBuffers.Put(out)
-	}()
+	var out *bytes.Buffer
+
+	if pool != nil {
+		outBuf := pool.Get(pool.MaxSize())[:0]
+		defer func() {
+			pool.Put(outBuf)
+		}()
+		out = bytes.NewBuffer(outBuf)
+	} else {
+		out = byteBuffers.Get().(*bytes.Buffer)
+		out.Reset()
+		defer byteBuffers.Put(out)
```

Comment:
can't we always pass a pool into here? i mean we can avoid using byteBuffers compltetely and use the more optimized bucketed pool